cmd2/utils.py

```diff
@@ -246,3 +246,57 @@ def natural_sort(list_to_sort: List[str]) -> List[str]:
     :return: the list sorted naturally
     """
     return sorted(list_to_sort, key=natural_keys)
+
+
+class StdSim(object):
+    """Class to simulate behavior of sys.stdout or sys.stderr.
+
+    Stores contents in internal buffer and optionally echos to the inner stream it is simulating.
+    """
+    class ByteBuf(object):
+        """Inner class which stores an actual bytes buffer and does the actual output if echo is enabled."""
+        def __init__(self, inner_stream, echo: bool = False) -> None:
+            self.byte_buf = b''
+            self.inner_stream = inner_stream
+            self.echo = echo
+
+        def write(self, b: bytes) -> None:
+            """Add bytes to internal bytes buffer and if echo is True, echo contents to inner stream."""
```

Comment:
To match stdout, this should raise a TypeError if bytes are not passed in.
Here are actual errors I received from sys.stdout.buffer.write().

- `sys.stdout.buffer.write('test string')`

> TypeError: a bytes-like object is required, not 'str'

- `sys.stdout.buffer.write(5)`

> TypeError: a bytes-like object is required, not 'int'

Reply:
StdSim now raises a TypeError exception if either write() method is passed the wrong type